src/uploaders.rs

```diff
@@ -148,7 +183,7 @@ impl Uploader {
             let mut body = Vec::new();
             LimitErrorReader::new(req.body(), maximums.max_upload_size).read_to_end(&mut body)?;
             verify_tarball(krate, vers, &body, maximums.max_unpack_size)?;
-            self.upload(&app.http_client()?, &path, body, "application/x-tar")?
+            self.upload(&path, body, "application/x-tar")?
```

Comment:
The removed call to `app.http_client()` is part of the test infrastructure to record and play back HTTP requests.  It looks like `upload` needs to add the proxy information as part of building the `PutObjectRequest`.